api/genomics/wdl/WgsCohortExtractPrep.wdl

```diff
@@ -78,50 +128,173 @@ task CreateCohortTable {
+    String fq_cohort_extract_table = read_string("fq_cohort_extract_table.txt")
   }
 
   runtime {
     memory: "3.75 GB"
     bootDiskSizeGb: "15"
     disks: "local-disk 10 HDD"
     preemptible: 3
-    # Temporary GATK that I'm using to get unblocked. We will realign with the mainstream GATK ah_var_store branch
-    # once they incorporate my changes.
     docker: "gcr.io/all-of-us-workbench-test/variantstore-extract-prep:2"
   }
-}
+} 
+
+task ExtractTask {
+    # indicates that this task should NOT be call cached
+    meta {
+       volatile: true
+    }
+
+    input {
+        # ------------------------------------------------
+        # Input args:
+        File reference
+        File reference_index
+        File reference_dict
+    
+        String fq_sample_table
+
+        File intervals
+
+        String fq_cohort_extract_table
+        String read_project_id
+        String output_file
+        String output_gcs_dir
+        
+        String? fq_filter_set_table
+        String? filter_set_name
+        
+        # Runtime Options:
+        File? gatk_override
+        
+        Int? local_sort_max_records_in_ram = 5000000
+    }
+    
+    String outdir = sub(output_gcs_dir, "/$", "")
+
+
+    # ------------------------------------------------
+    # Run our command:
+    command <<<
+        set -e
+        export GATK_LOCAL_JAR=~{default="/root/gatk.jar" gatk_override}
+
+        bq query --project_id ~{read_project_id} --max_rows=10000000 --format=csv --nouse_legacy_sql \
+          'select sample_id, sample_name FROM `~{fq_sample_table}`' | sed -e 1d \
+          > sample_map.csv
+
+        cat sample_map.csv
+
+        gatk --java-options "-Xmx9g" \
+            ExtractCohort \
+                --mode GENOMES --ref-version 38 --query-mode LOCAL_SORT \
+                -R "~{reference}" \
+                -O "~{output_file}" \
+                --local-sort-max-records-in-ram ~{local_sort_max_records_in_ram} \
+                --sample-file sample_map.csv \
+                --cohort-extract-table ~{fq_cohort_extract_table} \
+                -L ~{intervals} \
+                --project-id ~{read_project_id} \
+                ~{"--variant-filter-table " + fq_filter_set_table} \
+                ~{"--filter-set-name " + filter_set_name}
+        
+        gsutil cp ~{output_file} ~{output_gcs_dir}/
+        gsutil cp ~{output_file}.tbi ~{output_gcs_dir}/
+    >>>
+
+    # ------------------------------------------------
+    # Runtime settings:
+    runtime {
+        docker: "us.gcr.io/broad-gatk/gatk:4.1.8.0"
+        memory: "10 GB"
+        disks: "local-disk 100 HDD"
+        bootDiskSizeGb: 15
+        preemptible: 3
+        cpu: 2
+    }
+
+    # ------------------------------------------------
+    # Outputs:
+    output {
+        String sample_csv = "sample_map.csv"
+        File output_vcf = "~{output_file}"
+        File output_vcf_index = "~{output_file}.tbi"
+    }
+ }
+
+ task SplitIntervals {
```

Comment:
Is there anything new to review in this WDL? Or is this mostly just backfilling in tasks that were already implemented in the upstream WDL?